### PR TITLE
Move auth logic to own file

### PR DESCRIPTION
**Pull request by tbille**
Here the commit says first step because I still need to find a way to reduce the code in the function get_account.
I think we can remove some logic from there so that the auth process doesn't take half of the functions

### CODE REVIEW

File: authentication.py
Comment (nottrobin) [inline authentication.py]: one more new line

File: app.py
Comment (nottrobin) [inline app.py]: This line is too long

File: authentication.py
Comment (nottrobin) [inline authentication.py]: "Imported but unused"

File: authentication.py
Comment (nottrobin) [inline authentication.py]: Would you mind adding a docstring to this function?

File: authentication.py
Comment (nottrobin) [inline authentication.py]: Would you mind adding a docstring to describe this function?

**Review by nottrobin — APPROVED** [review]
Works and basically looks good.

A few PEP8 style points, and since you're refactoring, please also add docstrings to describe all functions.

Once that's done this is good to merge.

File: authentication.py
Comment (tbille) [reply authentication.py]: done but github didn.t marked as outdated

Comment (nottrobin) [reply authentication.py]: :+1:

File: authentication.py
Comment (nottrobin) [inline authentication.py]: Why so indented?

**Comment by tbille** [comment]
Please don't merge, I will squash the commits once it's approved

File: authentication.py
Comment (Lukewh) [inline authentication.py]: rename to `url_login` for consistency.

File: authentication.py
Comment (Lukewh) [inline authentication.py]: Maybe return a consistent object:
```
{
  status_code: number,
  redirect: string,
  reason: string
}

{
  status_code: 200,
  redirect: false,
  reason: false
}

{
  status_code: 307,
  redirect: login_url,
  reason: 'Need login'
}
```

**Review by Lukewh — CHANGES_REQUESTED** [review]
Looks good just a couple of things for consideration

File: app.py
Comment (bartaz) [inline app.py]: Not sure if this part is necessary (apart from debugging), and 'connected' name may be confusing.

**Review by bartaz — APPROVED** [review]

File: app.py
Comment (tbille) [reply app.py]: This is not the best, but it's not something that is going to be copied past to other projects.

File: authentication.py
Comment (nottrobin) [inline authentication.py]: This long function should really have a docstring, and it would be good if you could also add them to the other functions that still don't have them.

File: app.py
Comment (nottrobin) [reply app.py]: Yeah it's not ideal but it's not part of this PR or the library code.

File: authentication.py
Comment (tbille) [reply authentication.py]: done

**Review by Lukewh — APPROVED** [review]
Looks good thanks @tbille 👍

**Comment by nottrobin** [comment]
Me too :+1:

**Pull request merged**
